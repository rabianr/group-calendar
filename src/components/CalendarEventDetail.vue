<script setup lang="ts">
import { ref } from 'vue'
import { useI18n } from 'vue-i18n'
import { useCalendarStore } from '@/stores/calendar'
import type { DisplayEvent } from './CalendarEvent.types'
import IconTrash3 from './IconTrash3.vue'

const props = defineProps<{
  data: DisplayEvent
}>()

const { t } = useI18n()
const tableRef = ref()
const calendar = useCalendarStore()

defineExpose({
  tableRef
})

const deleteEvent = () => {
  const confirmed = confirm(t('Are you sure you want to delete this event?'))

  if (confirmed) {
    calendar.deleteEvent(props.data.memberName!, props.data.id!)
  }
}
</script>

<template>
  <table ref="tableRef" class="table-event-detail">
    <tbody>
      <tr>
        <th class="align-top text-end pe-2">{{ $t('Date') }} :</th>
        <td>
          {{ data.start }} <template v-if="data.start != data.end">ï½ž {{ data.end }} </template>
        </td>
      </tr>
      <tr>
        <th class="align-top text-end pe-2">{{ $t('Detail') }} :</th>
        <td>{{ data.detail }}</td>
      </tr>
      <tr>
        <th></th>
        <td>
          <button type="button" class="btn btn-link text-danger p-0" @click="deleteEvent">
            <IconTrash3 />
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</template>

<style lang="scss" scoped>
.table-event-detail {
  width: 100%;

  th {
    width: 60px;
    color: var(--bs-secondary);
  }

  td {
    white-space: pre-line;
  }
}
</style>
