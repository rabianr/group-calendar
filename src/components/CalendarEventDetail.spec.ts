import { describe, expect, it, vi } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'
import CalendarEventDetail from './CalendarEventDetail.vue'

describe('CalendarEventDetail コンポーネント', () => {
  const data = {
    title: 'テスト',
    detail: 'テスト内容',
    start: '2024-04-10',
    end: '2024-04-10',
    colspan: 1,
    pos: 0,
    last: false
  }

  it('スナップショットテスト', async () => {
    const wrapper = shallowMount(CalendarEventDetail, {
      global: {
        plugins: [createTestingPinia({ createSpy: vi.fn })]
      },
      props: {
        data
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ data: { ...data, end: '2024-04-11', colspan: 2 } })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
