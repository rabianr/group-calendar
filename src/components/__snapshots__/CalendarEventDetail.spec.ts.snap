// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CalendarEventDetail コンポーネント > スナップショットテスト 1`] = `
"<table data-v-3a649ac1="" class="table-event-detail">
  <tbody data-v-3a649ac1="">
    <tr data-v-3a649ac1="">
      <th data-v-3a649ac1="" class="align-top text-end pe-2">日付 :</th>
      <td data-v-3a649ac1="">2024-04-10
        <!--v-if-->
      </td>
    </tr>
    <tr data-v-3a649ac1="">
      <th data-v-3a649ac1="" class="align-top text-end pe-2">内容 :</th>
      <td data-v-3a649ac1="">テスト内容</td>
    </tr>
    <tr data-v-3a649ac1="">
      <th data-v-3a649ac1=""></th>
      <td data-v-3a649ac1=""><button data-v-3a649ac1="" type="button" class="btn btn-link text-danger p-0">
          <icon-trash3-stub data-v-3a649ac1=""></icon-trash3-stub>
        </button></td>
    </tr>
  </tbody>
</table>"
`;

exports[`CalendarEventDetail コンポーネント > スナップショットテスト 2`] = `
"<table data-v-3a649ac1="" class="table-event-detail">
  <tbody data-v-3a649ac1="">
    <tr data-v-3a649ac1="">
      <th data-v-3a649ac1="" class="align-top text-end pe-2">日付 :</th>
      <td data-v-3a649ac1="">2024-04-10 ～ 2024-04-11</td>
    </tr>
    <tr data-v-3a649ac1="">
      <th data-v-3a649ac1="" class="align-top text-end pe-2">内容 :</th>
      <td data-v-3a649ac1="">テスト内容</td>
    </tr>
    <tr data-v-3a649ac1="">
      <th data-v-3a649ac1=""></th>
      <td data-v-3a649ac1=""><button data-v-3a649ac1="" type="button" class="btn btn-link text-danger p-0">
          <icon-trash3-stub data-v-3a649ac1=""></icon-trash3-stub>
        </button></td>
    </tr>
  </tbody>
</table>"
`;
